internal/blobtesting/map.go

```diff
@@ -98,8 +105,13 @@ func (s *mapStorage) DeleteBlob(ctx context.Context, id blob.ID) error {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
+	if r, ok := s.retentionTime[id]; ok && r.After(s.timeNow()) {
+		return errors.New("cannot delete object before retention period expires")
+	}
+
 	delete(s.data, id)
 	delete(s.keyTime, id)
+	delete(s.retentionTime, id)
```

Comment:
The blob should not be deleted here, instead a deletion marker should be added.

Reply:
yes, I thought about it but wasn't sure about the required guarantees around the api behavior around map storage

if this is needed then I should really take this change in a separate pr as you proposed